source/include/ble/cs_CrownstoneCentral.h

```diff
@@ -0,0 +1,200 @@
+	 * @return ERR_WAIT_FOR_SUCCESS    When disconnecting. Wait for EVT_BLE_CENTRAL_DISCONNECTED.
+	 */
+	cs_ret_code_t disconnect();
+
+	/**
+	 * Write a control command, and get the result.
+	 *
+	 * When the result code in the result data is ERR_WAIT_FOR_SUCCESS, you will get another EVT_CS_CENTRAL_WRITE_RESULT event.
+	 *
+	 * TODO: add timeout.
+	 *
+	 * @param[in] commandType          What control command.
+	 * @param[in] data                 Pointer to the command payload data.
+	 * @param[in] size                 Size of the command payload data.
+	 *
+	 * @return ERR_WAIT_FOR_SUCCESS    When the write is started. Wait for EVT_CS_CENTRAL_WRITE_RESULT.
+	 */
+	cs_ret_code_t write(cs_control_cmd_t commandType, uint8_t* data, uint16_t size);
+
+	/**
+	 * Request the write buffer, to avoid having to copy data to the write buffer.
+	 *
+	 * @return     When busy:          A null pointer.
+	 * @return     On success:         A pointer to the write buffer, and the length of the buffer.
+	 */
+	cs_data_t requestWriteBuffer();
+
+private:
+	enum ServiceIndex {
+		SERVICE_INDEX_CROWNSTONE = 0,
+		SERVICE_INDEX_SETUP = 1,
+		SERVICE_INDEX_DEVICE_INFO = 2,
+		SERVICE_INDEX_DFU = 3,
+	};
+
+	enum class Operation: uint8_t {
+		NONE,
+		CONNECT,
+		WRITE
+	};
+
+	enum class ConnectSteps: uint8_t {
+		NONE = 0,
+		GET_ADDRESS,
+		CONNECT,
+		DISCOVER,
+		ENABLE_NOTIFICATIONS,
+		SESSION_KEY,
+		SESSION_DATA,
+		DONE
+	};
+
+	enum class WriteControlSteps: uint8_t {
+		NONE = 0,
+		WRITE,
+		RECEIVE_RESULT,
+		DONE
+	};
+
+	UUID _uuids[4];
+
+	uint16_t _sessionKeyHandle;
+	uint16_t _sessionDataHandle;
+	uint16_t _controlHandle;
+	uint16_t _resultHandle;
+	uint16_t _resultCccdHandle;
+	OperationMode _opMode = OperationMode::OPERATION_MODE_UNINITIALIZED;
+	Operation _currentOperation = Operation::NONE;
+	uint8_t _currentStep;
+
+	uint8_t _notificationNextIndex = 0;
+	uint16_t _notificationMergedDataSize = 0;
+
+	/**
+	 * Reset connection variables.
+	 */
+	void reset();
+
+	/**
+	 * Reset notification merger variables.
+	 */
+	void resetNotifactionMergerState();
+
+	/**
+	 * Enable notification on the result characteristic.
+	 * Sets current step.
+	 */
+	void enableNotifications();
+
+	/**
+	 * Read session key or session data, depending on operation mode.
+	 * Sets current step.
+	 */
+	void readSessionData();
+
+	/**
+	 * Merge this notification data with that of others.
+	 *
+	 * @param[in] notificationData     The data of this notification.
+	 * @param[out] resultData          The result data, only set on success.
+	 *
+	 * @return ERR_SUCCESS             All notifications have been merged, and decrypted. The resultData is set.
+	 * @return ERR_WAIT_FOR_SUCCESS    The notification has been merged, but there are more to come.
+	 * @return ERR_*                   The notification data could not be processed.
+	 */
+	cs_ret_code_t mergeNotification(const cs_const_data_t& notificationData, cs_data_t& resultData);
```

Comment:
Maybe a short explanation of what merging notifications exactly means?

Reply:
Added explanation in the top comment.